tools/setup-macos.sh

```
@@ -0,0 +1,125 @@
+        echo "Install Homebrew from https://brew.sh, then re-run this script."
+        exit 1
+    fi
+    echo "p7zip not found. Installing via Homebrew..."
+    brew install p7zip
+}
+
+check_deps() {
+    if ! command -v cargo &>/dev/null; then
+        # cargo may exist but not be on PATH yet (e.g. fresh rustup install)
+        if [[ -f "$HOME/.cargo/env" ]]; then
+            source "$HOME/.cargo/env"
```

Shellcheck (0.11.0)

[info] Not following: ./.cargo/env was not specified as input (see shellcheck -x).

(SC1091)